File: frontend/src/api/accountAnalysis.ts
import { api } from './client'

export type AnalysisJobStage =
  | 'queued'
  | 'fetching_games'
  | 'scanning_metadata'
  | 'analyzing_candidates'
  | 'deep_reviewing_examples'
  | 'saving_report'
  | 'complete'
  | 'failed'
  | 'cancelled'

export type AnalysisJobStatus = 'queued' | 'running' | 'complete' | 'failed' | 'cancelled'

export interface AnalysisJob {
  id: number
  status: AnalysisJobStatus
  stage: AnalysisJobStage
  progress_pct: number
  account_scope: Record<string, unknown>
  filters: Record<string, unknown>
  requested_game_ids: string[]
  completed_game_ids: string[]
  error: string | null
  report_id: number | null
  created_at: string
  updated_at: string
  started_at: string | null
  finished_at: string | null
}

export interface TrainingPlanFocus {
  category: string
  title: string
  summary: string
  habit: string[]
  confidence: 'trend_signal' | 'verified_examples'
}

export interface ReviewMoment {
  id?: string | null
  example_id?: string | null
  lesson_id?: string | null
  game_id: number
  platform_game_id: string | null
  platform: 'chesscom' | 'lichess' | string
  opponent: string | null
  result: 'W' | 'L' | 'D' | string | null
  time_control: string | null
  segment: string
  move_number: number
  color: 'white' | 'black'
  move_played: string
  played_san?: string | null
  fen_before?: string | null
  fen_after?: string | null
  better_move_san?: string | null
  better_move_uci?: string | null
  eval_loss_cp?: number | null
  win_pct_loss?: number | null
  verification_method?: string | null
  verified?: boolean
  theme_facts?: string[]
  practice_prompt?: string | null
  title: string
  coach_note: string
  pgn: string
}

export interface LessonContext {
  id?: string
  category?: string
  title?: string
  report_title?: string
  summary?: string
  habit?: string[]
  ai_context?: string
  practice_prompt?: string
  example_count?: number
}

export interface TrainingPlanReport {
  id: number
  created_at: string
  source_platforms: string[]
  scanned_range: Record<string, unknown>
  scan_summary: {
    eligible_games?: number
    parsed_games?: number
    candidate_positions?: number
    minimum_lesson_games?: number
    sample_status?: string
    verification_method?: string
    engine_note?: string
    result_counts?: Record<string, number>
  }
  time_control_breakdown: Array<{ segment: string; games: number }>
  top_trends: Array<{
    category: string
    label: string
    count: number
    confidence: 'trend_signal' | 'verified_examples'
    segments: Record<string, number>
  }>
  current_focus: TrainingPlanFocus
  review_moments: ReviewMoment[]
  opening_context: Array<{ opening: string; color?: 'white' | 'black' | string; games: number }>
  lesson_context?: LessonContext
  verified_examples?: ReviewMoment[]
  quality_summary?: Record<string, unknown>
  technical_evidence: Record<string, unknown>
}

export async function startTrainingPlanJob(): Promise<{ job: AnalysisJob; active_existing: boolean }> {
  return api.post('/account-analysis/jobs', {
    max_games: 500,
    months: 12,
    min_initial_seconds: 300,
  }, { timeoutMs: 30000 })
}

export async function getLatestTrainingPlanJob(): Promise<AnalysisJob | null> {
  return api.get('/account-analysis/jobs/latest')
}

export async function getTrainingPlanJob(jobId: number): Promise<{ job: AnalysisJob; report: TrainingPlanReport | null }> {
  return api.get(`/account-analysis/jobs/${jobId}`)
}

export async function cancelTrainingPlanJob(jobId: number): Promise<AnalysisJob> {
  return api.post(`/account-analysis/jobs/${jobId}/cancel`)
}

export async function retryTrainingPlanJob(jobId: number): Promise<AnalysisJob> {
  return api.post(`/account-analysis/jobs/${jobId}/retry`)
}

export async function getLatestTrainingPlanReport(): Promise<TrainingPlanReport | null> {
  const response = await api.get<{ report: TrainingPlanReport | null }>('/account-analysis/reports/latest')
  return response.report
}

File: backend/app/schemas/account_analysis.py
"""Schemas for account-wide training plan reports."""
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field

JobStage = Literal[
    "queued",
    "fetching_games",
    "scanning_metadata",
    "analyzing_candidates",
    "deep_reviewing_examples",
    "saving_report",
    "complete",
    "failed",
    "cancelled",
]
JobStatus = Literal["queued", "running", "complete", "failed", "cancelled"]


class TrainingPlanFocus(BaseModel):
    category: str
    title: str
    summary: str
    habit: list[str] = Field(default_factory=list)
    confidence: Literal["trend_signal", "verified_examples"]


class ReviewMoment(BaseModel):
    id: str | None = None
    example_id: str | None = None
    lesson_id: str | None = None
    game_id: int
    platform_game_id: str | None
    platform: str
    opponent: str | None
    result: str | None
    time_control: str | None
    segment: str
    move_number: int
    color: Literal["white", "black"]
    move_played: str
    played_san: str | None = None
    fen_before: str | None = None
    fen_after: str | None = None
    better_move_san: str | None = None
    better_move_uci: str | None = None
    eval_loss_cp: int | None = None
    win_pct_loss: float | None = None
    verification_method: str | None = None
    verified: bool = False
    theme_facts: list[str] = Field(default_factory=list)
    practice_prompt: str | None = None
    title: str
    coach_note: str
    pgn: str


class TrainingPlanReport(BaseModel):
    id: int
    created_at: datetime
    source_platforms: list[str]
    scanned_range: dict[str, Any]
    scan_summary: dict[str, Any]
    time_control_breakdown: list[dict[str, Any]]
    top_trends: list[dict[str, Any]]
    current_focus: TrainingPlanFocus
    review_moments: list[ReviewMoment]
    opening_context: list[dict[str, Any]]
    technical_evidence: dict[str, Any]
    lesson_context: dict[str, Any] = Field(default_factory=dict)
    verified_examples: list[ReviewMoment] = Field(default_factory=list)
    quality_summary: dict[str, Any] = Field(default_factory=dict)


class AnalysisJobResponse(BaseModel):
    id: int
    status: JobStatus
    stage: JobStage
    progress_pct: int
    account_scope: dict[str, Any]
    filters: dict[str, Any]
    requested_game_ids: list[str]
    completed_game_ids: list[str]
    error: str | None
    report_id: int | None
    created_at: datetime
    updated_at: datetime
    started_at: datetime | None
    finished_at: datetime | None

    model_config = {"from_attributes": True}


class StartAnalysisRequest(BaseModel):
    max_games: int = Field(default=500, ge=25, le=500)
    months: int = Field(default=12, ge=1, le=24)
    min_initial_seconds: int = Field(default=300, ge=60, le=3600)
    platforms: list[Literal["chesscom", "lichess"]] | None = None


class StartAnalysisResponse(BaseModel):
    job: AnalysisJobResponse
    active_existing: bool


class LatestReportResponse(BaseModel):
    report: TrainingPlanReport | None


class JobReportResponse(BaseModel):
    job: AnalysisJobResponse
    report: TrainingPlanReport | None
